### PR TITLE
Refactor simple options

### PR DESCRIPTION
**Pull request by fi-ya**
This ticket focuses on handling OPTIONS routes, '/method_options' and 'method_options2' using test driven development with changes made to the following classes. 
- 'Router' - update 'createResourceAndHandlerMap' method by assigning requested target the correct handle
- 'OptionsHandler' and OptionsHandlerTwo' - test allowedMethods and response are correct 

Acceptance test: 27 successful (Simple GET, Simple HEAD, Simple OPTIONS, Method Not Allowed)

### CODE REVIEW

File: src/main/java/org/httpserver/handler/OptionsHandler.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class OptionsHandler implements Handler {
+    @Override
+    public List<String> allowedHttpMethods() {
+        return Arrays.asList(HttpMethod.GET.getHttpMethod(), HttpMethod.HEAD.getHttpMethod(), HttpMethod.OPTIONS.getHttpMethod());
+    }
+
+
+    @Override
+    public Response handleResponse(Request request) {
+        String CRLF = "\r\n";
```
Comment (daddyshortlegs) [inline src/main/java/org/httpserver/handler/OptionsHandler.java:22]: You've got this declared 5 times in your code base. Time to extract it and make it a constant.

File: src/main/java/org/httpserver/handler/OptionsHandler.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        ResponseBuilder responseBuilder = new ResponseBuilder();
+        return responseBuilder.buildResponse(responseStatusLine, responseHeaders, responseBody);
+    }
+
+    private String handleStatusLine(Request request) {
+        String SP = " ";
+        String statusCode = StatusCode.OK.getStatusCode();
+        String statusText = String.valueOf(StatusCode.OK);
+
+        return request.getHttpVersion() + SP + statusCode + SP + statusText;
```
Comment (daddyshortlegs) [inline src/main/java/org/httpserver/handler/OptionsHandler.java:37]: Wouldn't it be nice if Java had string interpolation. Alas it does not :-(
However, you could use String.format here to build your string.

File: src/main/java/org/httpserver/handler/OptionsHandlerTwo.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+public class OptionsHandlerTwo implements Handler {
+
+    @Override
+    public List<String> allowedHttpMethods() {
+        return Arrays.asList(HttpMethod.GET.getHttpMethod(), HttpMethod.HEAD.getHttpMethod(), HttpMethod.OPTIONS.getHttpMethod(), HttpMethod.POST.getHttpMethod(), HttpMethod.PUT.getHttpMethod());
+    }
+
+    @Override
+    public Response handleResponse(Request request) {
+        String CRLF = "\r\n";
```
Comment (daddyshortlegs) [inline src/main/java/org/httpserver/handler/OptionsHandlerTwo.java:21]: Aarggh, another CRLF 🤣

File: src/test/java/org/httpserver/handler/OptionsHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        OptionsHandler optionsHandler = new OptionsHandler();
+
+        assertTrue(optionsHandler.allowedHttpMethods().contains("GET"));
+        assertTrue(optionsHandler.allowedHttpMethods().contains("HEAD"));
+        assertTrue(optionsHandler.allowedHttpMethods().contains("OPTIONS"));
+        assertEquals(3, optionsHandler.allowedHttpMethods().size());
+    }
+
+    @Test
+    void returnsResponseWithStatusLineHeadersAndEmptyBody_whenMethodOptions() {
```
Comment (daddyshortlegs) [inline src/test/java/org/httpserver/handler/OptionsHandlerTest.java:25]: Nice name

**Review by daddyshortlegs — COMMENTED** [review]
Nothing untoward here that I can see, mainly just a bit of cleanliness

File: src/main/java/org/httpserver/handler/OptionsHandler.java (hunk 1 above)
Comment (fi-ya) [reply src/main/java/org/httpserver/handler/OptionsHandler.java:22]: i have created a class at the root level called `Constant` like this
```
public class Constant {
    public static String CRLF = "\r\n";
    public static String SP = " ";
}
```

File: src/main/java/org/httpserver/handler/OptionsHandler.java (hunk 2 above)
Comment (fi-ya) [reply src/main/java/org/httpserver/handler/OptionsHandler.java]: Thanks, String.format is pretty handy, I dont need the SP variable anymore. I did the following 
```
  private String handleStatusLine(Request request) {
        String httpVersion = request.getHttpVersion();
        String statusCode = StatusCode.OK.getStatusCode();
        String statusText = String.valueOf(StatusCode.OK);

        return String.format("%s %s %s", httpVersion, statusCode, statusText);
    }
    ```

File: src/test/java/org/httpserver/handler/OptionsHandlerTest.java (hunk 4 above)
Comment (fi-ya) [reply src/test/java/org/httpserver/handler/OptionsHandlerTest.java:25]: Thanks!

**Pull request merged**